### PR TITLE
Add validation for CreateGrant and CreateFact

### PR DESCRIPTION
**Pull request by zhouzhuojie**

### CODE REVIEW

File: pkg/apimodel/grant.go [hunk 1]
```diff
@@ -2,16 +2,16 @@ package apimodel
 
 // CreateGrant represents the request object of creating a grant
 type CreateGrant struct {
-	Domain         string   `json:"domain"`
-	AllowedActions []string `json:"allowed_actions"`
+	Domain             string   `json:"domain" validate:"fqdn"`
```
Comment (jperichon) [inline pkg/apimodel/grant.go:5]: To triple check, will this allow *?

Comment (zhouzhuojie) [reply pkg/apimodel/grant.go:5]: I was thinking of making it the same as cookie sharing cross-domain rule.

E.g.

```
example.com
```

gives you the access to 
```
foo.example.com
bar.example.com
```

And still makes it compatible as https://en.wikipedia.org/wiki/Fully_qualified_domain_name

File: pkg/config/env.go [hunk 2]
```diff
@@ -45,7 +45,7 @@ var ENV = struct {
 	GrantJWTSigningKey    string `env:"OPV_GRANT_JWT_SIGNING_KEY" envDefault:"please_change_to_random_32bytes"`
 	GrantJWTValidationKey string `env:"OPV_GRANT_JWT_VALIDATION_KEY" envDefault:"please_change_to_random_32bytes"`
 
-	AuthzCasbinAutoloadInterval time.Duration `env:"OPV_AUTHZ_CASBIN_AUTOLOAD_INTERVAL" envDefault:"30s"`
+	AuthzCasbinAutoloadInterval time.Duration `env:"OPV_AUTHZ_CASBIN_AUTOLOAD_INTERVAL" envDefault:"3s"`
```
Comment (zhouzhuojie) [inline pkg/config/env.go:48]: btw, changed to `3s` to help localhost dev default @jperichon

File: pkg/apimodel/grant.go (hunk 1 above)
Comment (zhouzhuojie) [reply pkg/apimodel/grant.go:5]: `*` is no not valid, the regex is defined here https://github.com/go-playground/validator/blob/d6b17fd90bd4dd9d16a594c3035ceadc3de0193a/regexes.go#L41

**Review by jperichon — APPROVED** [review]

**Pull request merged**